daemon/src/db.rs

```diff
@@ -246,159 +232,205 @@ pub async fn load_cfd_by_order_id(
     order_id: OrderId,
     conn: &mut PoolConnection<Sqlite>,
 ) -> Result<Cfd> {
-    let order_uuid = serde_json::to_string(&order_id)?;
-
     let row = sqlx::query!(
         r#"
-        select
-            orders.uuid as order_id,
-            orders.initial_price as price,
-            orders.min_quantity as min_quantity,
-            orders.max_quantity as max_quantity,
-            orders.leverage as leverage,
-            orders.trading_pair as trading_pair,
-            orders.position as position,
-            orders.origin as origin,
-            orders.liquidation_price as liquidation_price,
-            orders.creation_timestamp as creation_timestamp,
-            orders.term as term,
-            orders.oracle_event_id,
-            cfds.quantity_usd as quantity_usd,
-            cfd_states.state as state
-        from cfds as cfds
-        inner join orders as orders on cfds.order_id = orders.id
-        inner join cfd_states as cfd_states on cfd_states.cfd_id = cfds.id
-        where cfd_states.state in (
+        with ord as (
             select
-              state
-              from cfd_states
-            where cfd_id = cfds.id
-            order by id desc
-            limit 1
+                id as order_id,
+                uuid,
+                trading_pair,
+                position,
+                initial_price,
+                min_quantity,
+                max_quantity,
+                leverage,
+                liquidation_price,
+                creation_timestamp_seconds as ts_secs,
+                creation_timestamp_nanoseconds as ts_nanos,
+                term_seconds as term_secs,
+                term_nanoseconds as term_nanos,
+                origin,
+                oracle_event_id
+            from orders
+        ),
+
+        cfd as (
+            select
+                ord.order_id,
+                id as cfd_id,
+                quantity_usd
+            from cfds
+                inner join ord on ord.order_id = id
+        ),
+
+        state as (
```

Comment:
nit: I think this should be called `cfds_with_latest_state` - because in my understanding that is what this represents.

Reply:
was this not actioned on purpose? would be nice to say so. Resolved for me means that it was actioned.

Reply:
While there is no general style guide that I am aware of for SQL in general (could be well wrong about this, I'm no expert!) there certainly isn't one for us internally. That being said, in practice most SQL aliases end up with really useful aliases like `a`, `b`, etc., which obviously suck but in my experience shorter phrases > longer ones as they tend to compound and things get progressively less readable with larger, more complex queries (which this most certainly is not, of course).

Thus, my action was to ignore the suggestion as I simply disagree with it is all; I think `state` is good alias as it captures the intent of the sub-query, that being that it is getting the "state" part of the overall query.

Apologies if you disagree, I read the above as a suggestion only is all, and since it didn't affect anything beyond stylistic concerns I chose to keep my original wording.

Reply:
all good, would just be nice to add a quick comment next time and then resolve the comment - then I know that it was on purpose and not just accidentally resolved.